### PR TITLE
Feature - collab perms

### PR DESCRIPTION
**Pull request by joefitter**
* added collab:edit user to project edit
* added collab:edit user to training read
* added new project:submit permission

### CODE REVIEW

File: lib/utils/allowed.js [hunk 1]
```diff
@@ -166,14 +166,15 @@ function roleIsAllowed({ db, model, permission, user: unscoped, subject = {} })
...
+        const permLevel = pieces[2];
         const id = subject.projectId || subject.id;
         if (!id) {
           return false;
         }
         const { Project, Profile } = db;
         return Promise.resolve()
           .then(() => Project.queryWithDeleted()
-            .whereIsCollaborator(user.id)
+            .whereIsCollaborator(user.id, permLevel)
```
Comment (lennym) [inline lib/utils/allowed.js:177]: I didn't see the schema PR, but will this be a no-op for `project:collaborator` perms where the level is undefined?

File: config.js [hunk 2]
```diff
@@ -75,7 +75,8 @@ module.exports = {
       transfer: ['project:own', 'asru:licensing'],
       import: ['establishment:*'],
       create: ['profile:own', 'establishment:admin'],
-      update: ['holdingEstablishment:admin', 'project:own', 'asru:licensing'],
+      update: ['holdingEstablishment:admin', 'project:own', 'asru:licensing', 'project:collaborator:edit'],
+      submit: ['holdingEstablishment:admin', 'project:own', 'asru:licensing'],
```
Comment (lennym) [inline config.js:79]: Can you rebase and add the equivalent to `project.rops`?

File: config.js [hunk 3]
```diff
@@ -75,7 +75,8 @@ module.exports = {
...
       rops: {
         update: ['asru:licensing']
```
Comment (lennym) [inline config.js:81]: Here ☝️

File: lib/utils/allowed.js (hunk 1 above)
Comment (joefitter) [reply lib/utils/allowed.js:177]: yes, it will just lookup without adding the role to the where

File: config.js (hunk 2 above)
Comment (joefitter) [reply config.js:79]: yep

File: config.js (hunk 3 above)
Comment (joefitter) [reply config.js:81]: 👍

File: config.js (hunk 2 above)
Comment (lennym) [reply config.js:79]: We probably/possibly also want the distinct `submit` permission.

File: config.js [hunk 4]
```diff
@@ -75,9 +75,10 @@ module.exports = {
       transfer: ['project:own', 'asru:licensing'],
       import: ['establishment:*'],
       create: ['profile:own', 'establishment:admin'],
-      update: ['holdingEstablishment:admin', 'project:own', 'asru:licensing'],
+      update: ['holdingEstablishment:admin', 'project:own', 'asru:licensing', 'project:collaborator:edit'],
+      submit: ['holdingEstablishment:admin', 'project:own', 'asru:licensing'],
       rops: {
-        update: ['establishment:admin', 'project:own', 'asru:licensing'],
+        update: ['establishment:admin', 'project:own', 'asru:licensing', 'project:collaborator:edit'],
```
Comment (lennym) [inline config.js:81]:
```suggestion
        update: ['establishment:admin', 'project:own', 'asru:licensing', 'project:collaborator:edit'],
        submit: ['establishment:admin', 'project:own', 'asru:licensing'],
```

**Pull request merged**